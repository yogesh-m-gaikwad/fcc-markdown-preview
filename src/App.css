.color-picker{
    height: 30px;
    width: 30px;
    border-radius: 5px;    
}

.btn {
    margin: 5px;
    outline: none;
    border: 1px solid lightgray;
}

body{
    overflow-x: hidden;
    background-color: #CFF09E;
    color: #0B486B;
}

#app {
    justify-content: center;
    align-items: center;
    height: auto;
    width: 100vw;
}

img {
    width: 80%;
}

#editor-wrapper {
    margin: 30px;
    padding: 10px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border: 1px solid rgba(121,189,154, .5);
    border-radius: 15px;
    box-shadow: 5px 15px 60px 20px rgba(121,189,154, .15);
    background-color: #E1F5C4;
}

#editor-wrapper textarea {
    width: 100%;
    padding: 20px;
    font-family: Consolas;
    font-size: 14px;
    box-sizing: border-box;    
    min-height: 402px;
    border-color: rgba(5, 5, 5, .1);
    border-radius: 15px;
    background-color: #A8DBA8;
}

#editor-wrapper textarea:focus, #editor textarea:active {
    outline: none;
}

#preview {
    margin: 30px;
    padding: 10px 25px;
    border: 1px solid rgba(121,189,154, .5);
    border-radius: 15px;
    box-shadow: 5px 15px 60px 20px rgba(121,189,154, .15);
    background-color: #E1F5C4;
}


/* Scorllbar CSS */
::-webkit-scrollbar {
    width: 10px;
}

::-webkit-scrollbar-track {
    background: rgba(59,34,34, 0.1);
}

::-webkit-scrollbar-thumb {
    background: #0B486B;
    border-radius: 5px;
}

::-webkit-scrollbar-thumb:hover {
    background: #3B8686;
}