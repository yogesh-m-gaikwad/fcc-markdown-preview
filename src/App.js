import React from 'react';
import { PreviewBox } from './features/markup/PreviewBox';
import { EditorBox } from './features/markup/EditorBox';
import { connect } from 'react-redux';
import { refreshAsync } from './features/markup/markupSlice';
import './App.css';

const mapStateToProps = (state) => ({
  text: state.markup.text
});

const mapDispatchToProps = (dispatch) => {
  return {
    refresh: refreshAsync,
    dispatch
  };
};

class App extends React.Component {
  constructor(props){
    super(props);
    console.log("App: created!");
  }

  render(){
    return (
      <div id="main-container" className="container">
          <div className="row">
              <EditorBox text={this.props.text} refresh={this.props.refresh}/>
          </div>
          <div className="row">
              <PreviewBox />
          </div>
      </div>
    );
  }
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(App);