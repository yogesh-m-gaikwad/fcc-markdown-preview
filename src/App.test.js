import React from 'react';
import { render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import store from './app/store';
import App from './App';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <Provider store={store}>
      <App />
    </Provider>, 
    div);
  
  ReactDOM.unmountComponentAtNode(div);
});

test('App loads', () => {
  const component = renderer.create(
    <Provider store={store}>
      <App />
    </Provider>,
  );
  let app = component.toJSON();
  expect(app).toMatchSnapshot();

  expect(app).toMatchSnapshot();
});