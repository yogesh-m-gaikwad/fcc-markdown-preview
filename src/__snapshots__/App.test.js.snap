// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App loads 1`] = `
<div
  className="container"
  id="main-container"
>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
      id="editor-wrapper"
    >
      <textarea
        defaultValue="Marked - Markdown Parser
========================

[Marked] lets you convert [Markdown] into HTML.
Markdown is a simple text format whose goal is to be very easy to read and write, even when not converted to HTML.  
This demo page will let you type anything you like and see how it gets converted.  Live.  No more waiting around.

How To Use The Demo
-------------------

1. Type in stuff on the left.
2. See the live updates on the right.

That's it.  Pretty simple.  There's also a drop-down option in the upper right to switch between various views:

- **Preview:**  A live display of the generated HTML as it would render in a browser.
- **HTML Source:**  The generated HTML before your browser makes it pretty.
- **Lexer Data:**  What [marked] uses internally, in case you like gory stuff like this.
- **Quick Reference:**  A brief run-down of how to format things using markdown.

Why Markdown?
-------------

It's easy.  It's not overly bloated, unlike HTML.  Also, as the creator of [markdown] says,

> The overriding design goal for Markdown's
> formatting syntax is to make it as readable
> as possible. The idea is that a
> Markdown-formatted document should be
> publishable as-is, as plain text, without
> looking like it's been marked up with tags
> or formatting instructions.

Ready to start writing?  Either start changing stuff on the left or
[clear everything](/demo/?text=) with a simple click.

[Marked]: https://github.com/markedjs/marked/
[Markdown]: http://daringfireball.net/projects/markdown/"
        id="editor"
        onChange={[Function]}
        style={
          Object {
            "color": "black",
          }
        }
      />
      <div
        className="btn-group"
        role="group"
      >
        <button
          aria-label={null}
          className="color-picker btn btn-primary"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "red",
              "color": "white",
            }
          }
          type="button"
        />
        <button
          aria-label={null}
          className="color-picker btn btn-primary"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "blue",
              "color": "white",
            }
          }
          type="button"
        />
        <button
          aria-label={null}
          className="color-picker btn btn-primary active"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "black",
              "color": "white",
            }
          }
          type="button"
        />
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h1 id=\\"marked---markdown-parser\\">Marked - Markdown Parser</h1>
<p><a href=\\"https://github.com/markedjs/marked/\\">Marked</a> lets you convert <a href=\\"http://daringfireball.net/projects/markdown/\\">Markdown</a> into HTML.
Markdown is a simple text format whose goal is to be very easy to read and write, even when not converted to HTML.<br>This demo page will let you type anything you like and see how it gets converted.  Live.  No more waiting around.</p>
<h2 id=\\"how-to-use-the-demo\\">How To Use The Demo</h2>
<ol>
<li>Type in stuff on the left.</li>
<li>See the live updates on the right.</li>
</ol>
<p>That&#39;s it.  Pretty simple.  There&#39;s also a drop-down option in the upper right to switch between various views:</p>
<ul>
<li><strong>Preview:</strong>  A live display of the generated HTML as it would render in a browser.</li>
<li><strong>HTML Source:</strong>  The generated HTML before your browser makes it pretty.</li>
<li><strong>Lexer Data:</strong>  What <a href=\\"https://github.com/markedjs/marked/\\">marked</a> uses internally, in case you like gory stuff like this.</li>
<li><strong>Quick Reference:</strong>  A brief run-down of how to format things using markdown.</li>
</ul>
<h2 id=\\"why-markdown\\">Why Markdown?</h2>
<p>It&#39;s easy.  It&#39;s not overly bloated, unlike HTML.  Also, as the creator of <a href=\\"http://daringfireball.net/projects/markdown/\\">markdown</a> says,</p>
<blockquote>
<p>The overriding design goal for Markdown&#39;s
formatting syntax is to make it as readable
as possible. The idea is that a
Markdown-formatted document should be
publishable as-is, as plain text, without
looking like it&#39;s been marked up with tags
or formatting instructions.</p>
</blockquote>
<p>Ready to start writing?  Either start changing stuff on the left or
<a href=\\"/demo/?text=\\">clear everything</a> with a simple click.</p>
",
        }
      }
      id="preview"
    />
  </div>
</div>
`;

exports[`App loads 2`] = `
<div
  className="container"
  id="main-container"
>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
      id="editor-wrapper"
    >
      <textarea
        defaultValue="Marked - Markdown Parser
========================

[Marked] lets you convert [Markdown] into HTML.
Markdown is a simple text format whose goal is to be very easy to read and write, even when not converted to HTML.  
This demo page will let you type anything you like and see how it gets converted.  Live.  No more waiting around.

How To Use The Demo
-------------------

1. Type in stuff on the left.
2. See the live updates on the right.

That's it.  Pretty simple.  There's also a drop-down option in the upper right to switch between various views:

- **Preview:**  A live display of the generated HTML as it would render in a browser.
- **HTML Source:**  The generated HTML before your browser makes it pretty.
- **Lexer Data:**  What [marked] uses internally, in case you like gory stuff like this.
- **Quick Reference:**  A brief run-down of how to format things using markdown.

Why Markdown?
-------------

It's easy.  It's not overly bloated, unlike HTML.  Also, as the creator of [markdown] says,

> The overriding design goal for Markdown's
> formatting syntax is to make it as readable
> as possible. The idea is that a
> Markdown-formatted document should be
> publishable as-is, as plain text, without
> looking like it's been marked up with tags
> or formatting instructions.

Ready to start writing?  Either start changing stuff on the left or
[clear everything](/demo/?text=) with a simple click.

[Marked]: https://github.com/markedjs/marked/
[Markdown]: http://daringfireball.net/projects/markdown/"
        id="editor"
        onChange={[Function]}
        style={
          Object {
            "color": "black",
          }
        }
      />
      <div
        className="btn-group"
        role="group"
      >
        <button
          aria-label={null}
          className="color-picker btn btn-primary"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "red",
              "color": "white",
            }
          }
          type="button"
        />
        <button
          aria-label={null}
          className="color-picker btn btn-primary"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "blue",
              "color": "white",
            }
          }
          type="button"
        />
        <button
          aria-label={null}
          className="color-picker btn btn-primary active"
          onClick={[Function]}
          style={
            Object {
              "backgroundColor": "black",
              "color": "white",
            }
          }
          type="button"
        />
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<h1 id=\\"marked---markdown-parser\\">Marked - Markdown Parser</h1>
<p><a href=\\"https://github.com/markedjs/marked/\\">Marked</a> lets you convert <a href=\\"http://daringfireball.net/projects/markdown/\\">Markdown</a> into HTML.
Markdown is a simple text format whose goal is to be very easy to read and write, even when not converted to HTML.<br>This demo page will let you type anything you like and see how it gets converted.  Live.  No more waiting around.</p>
<h2 id=\\"how-to-use-the-demo\\">How To Use The Demo</h2>
<ol>
<li>Type in stuff on the left.</li>
<li>See the live updates on the right.</li>
</ol>
<p>That&#39;s it.  Pretty simple.  There&#39;s also a drop-down option in the upper right to switch between various views:</p>
<ul>
<li><strong>Preview:</strong>  A live display of the generated HTML as it would render in a browser.</li>
<li><strong>HTML Source:</strong>  The generated HTML before your browser makes it pretty.</li>
<li><strong>Lexer Data:</strong>  What <a href=\\"https://github.com/markedjs/marked/\\">marked</a> uses internally, in case you like gory stuff like this.</li>
<li><strong>Quick Reference:</strong>  A brief run-down of how to format things using markdown.</li>
</ul>
<h2 id=\\"why-markdown\\">Why Markdown?</h2>
<p>It&#39;s easy.  It&#39;s not overly bloated, unlike HTML.  Also, as the creator of <a href=\\"http://daringfireball.net/projects/markdown/\\">markdown</a> says,</p>
<blockquote>
<p>The overriding design goal for Markdown&#39;s
formatting syntax is to make it as readable
as possible. The idea is that a
Markdown-formatted document should be
publishable as-is, as plain text, without
looking like it&#39;s been marked up with tags
or formatting instructions.</p>
</blockquote>
<p>Ready to start writing?  Either start changing stuff on the left or
<a href=\\"/demo/?text=\\">clear everything</a> with a simple click.</p>
",
        }
      }
      id="preview"
    />
  </div>
</div>
`;
